ui/shared/components/panel/view-pedigree-image/PedigreeImagePanel.jsx

```diff
@@ -1,133 +1,348 @@
+    const newParents = {}
+    dataset.filter(
+      ({ mother, father }) => (mother && !father) || ((!mother && father)),
+    ).forEach((indiv) => {
+      const placeholderId = `${indiv.mother || indiv.father}-spouse-placeholder`
+      if (!newParents[placeholderId]) {
+        newParents[placeholderId] = {
+          name: placeholderId,
+          sex: indiv.mother ? 'M' : 'F',
+          top_level: true,
+        }
+      }
+      if (indiv.mother) {
+        indiv.father = placeholderId
+      } else {
+        indiv.mother = placeholderId
+      }
+    })
+
+    return this.yobToAge([...dataset, ...Object.values(newParents)])
+  }
+
+  yobToAge = dataset => dataset.map(o => ({ ...o, age: o.yob && new Date().getFullYear() - o.yob }))
+
+  editIndividual = (opts, { data }) => {
+    this.setState({
+      editIndividual: {
+        data,
+        save: (newData) => {
+          Object.assign(data, newData)
+          this.redrawPedigree(opts)
+        },
+      } })
+    this.props.openIndividualModal()
+  }
 
-const mapDeleteDispatchToProps = (dispatch, ownProps) => {
-  return {
-    onSubmit: () => {
-      return dispatch(updateFamily({ familyField: 'pedigree_image', familyGuid: ownProps.familyGuid }))
-    },
-    onSuccess: () => {
-      dispatch(closeModal(ownProps.modalId))
-    },
+  getPedigreeDataset = () => {
+    return currentDataset(this.state.pedigreeOpts)
   }
 }
 
-const DeletePedigreeImageButton = connect(null, mapDeleteDispatchToProps)(BaseDeletePedigreeImageButton)
+const mapStateToProps = (state, ownProps) => ({
+  individuals: getIndividualsByFamily(state)[ownProps.family.familyGuid],
+})
+
+const mapDispatchToProps = dispatch => ({
+  openIndividualModal: () => {
+    dispatch(openModal(EDIT_INDIVIDUAL_MODAL_ID))
+  },
+})
+
+const PedigreeImage = connect(mapStateToProps, mapDispatchToProps)(BasePedigreeImage)
+
+// Do not crash the entire page if pedigree js is breaking
+const PedigreeError = () => <Icon name="picture" />
+const SafePedigreeImage = props =>
+  <ErrorBoundary FallbackComponent={PedigreeError}><PedigreeImage {...props} /></ErrorBoundary>
 
-const PedigreeImagePanel = React.memo((props) => {
-  if (!props.family.pedigreeImage) {
-    return props.isEditable && !props.compact ?
-      <EditPedigreeImageButton family={props.family} buttonText="Upload Pedigree" /> : null
+const PedigreeImagePanel = React.memo(({ family, isEditable, compact, disablePedigreeZoom }) => {
+  const hasPedImage = family.pedigreeImage || family.pedigreeDataset || family.individualGuids.length > 1
+  if (!hasPedImage && (!isEditable || compact)) {
+    return null
   }
-  const image = <PedigreeImage
-    src={props.family.pedigreeImage}
-    disablePedigreeZoom={props.disablePedigreeZoom}
-    compact={props.compact}
+
+  const image = hasPedImage && <SafePedigreeImage
+    family={family}
+    disablePedigreeZoom={disablePedigreeZoom}
+    maxHeight={compact ? '35' : '150'}
```

Comment:
`'150'` seems a little large. The pedigree image could interfere with the column next to it.

Reply:
this is what it was before and it looks right